server/game/cards/Time/FateReflection.js

```diff
@@ -0,0 +1,38 @@
+const { BattlefieldTypes } = require('../../../constants.js');
+const Card = require('../../Card.js');
+
+class FateReflection extends Card {
+    setupCardAbilities(ability) {
+        this.interrupt({
+            when: {
+                onDamageDealt: (event, context) =>
+                    event.context.player === context.player.opponent && // opponent action
+                    BattlefieldTypes.includes(event.card.type) && // its a unit
+                    !event.fightEvent // not a fight
+            },
+            gameAction: ability.actions.preventDamage((context) => ({
+                event: context.event,
+                amount: context.event.amount
+            })),
+            then: {
+                alwaysTriggers: true, // need to target first, then deal damage if I want targetting to appear
+                target: {
+                    activePromptTitle: "Choose an opponent's unit to receive the damage instead",
+                    cardType: BattlefieldTypes,
+                    controller: 'opponent',
+                    gameAction: ability.actions.dealDamage((context) => ({
+                        amount: context.preThenEvent.gameAction.amount, // needed to specify which amount method's amount to use
+                        showMessage: true
+                    }))
+                }
+            },
+            message:
+                '{0} plays {1} to prevent all damage to {2} then deals that damage to an opposing unit',
+            messageArgs: (context) => [context.player, context.source, context.event.card]
```

Comment:
with message, array items 0 1 and 2 are prefilled for you. anything you add in messageArgs starts at index 3, so nothing you have here would be used (but it looks similar to the defaults). is the logging all working correctly? i think the default for {2} is target, though not sure which target it refers to here. https://github.com/danj3000/ashteki/blob/master/server/game/ThenAbility.js#L31

Reply:
I just used the template for Angelic Rescue where the arguments are specificed in this fashion. It seemed that when I added a {3} I could display the damage that was dealt, but perhaps I didn't test it correctly.

Reply:
It's good as-is. I tested locally, and without messageArgs it doesn't work.
I also tweaked the tests before merging, so all is good.